### PR TITLE
zcash_client_sqlite: Rework accounts to support imported viewing keys

### PR DESCRIPTION
**Pull request by str4d**
Squashed version of #1175.

### CODE REVIEW

**Comment by str4d** [comment]
The current commit in this PR (0c80893ce0f8b88d5329157dfb583af14cd2b3aa) [is identical to](https://github.com/nerdcash/librustzcash/compare/cc6e7153676bf6db93eef91a3466de9623f7e51e..zcash:librustzcash:0c80893ce0f8b88d5329157dfb583af14cd2b3aa) the tip commit in #1175 (nerdcash/librustzcash@cc6e7153676bf6db93eef91a3466de9623f7e51e).

**Comment by AArnott** [comment]
@str4d did you see @nuttycom's comment here: https://github.com/zcash/librustzcash/pull/1175#pullrequestreview-1925276413

**Review by nuttycom — APPROVED** [review]
utACK bfc2278bfe38b792dc494ffd72c1dda3294dd8aa

This is identical to #1175, which was previously approved, with the addition of bugfixes and updating to use sapling-crypto 0.1.2 for IVK encoding.

File: zcash_client_sqlite/src/wallet.rs [hunk 1]
```diff
@@ -332,12 +332,8 @@ pub(crate) fn ufvk_to_uivk<P: consensus::Parameters>(
...
     }
     if let Some(sapling) = ufvk.sapling() {
-        let dk = sapling.to_bytes();
-        let ivk = sapling.to_ivk(Scope::External).to_repr();
-        let mut divk = [0u8; 64];
-        divk[..32].copy_from_slice(&dk[96..]);
-        divk[32..].copy_from_slice(&ivk[..]);
-        ivks.push(Ivk::Sapling(divk));
+        let ivk = sapling.to_external_ivk();
+        ivks.push(Ivk::Sapling(ivk.to_bytes()));
```
Comment (AArnott) [inline zcash_client_sqlite/src/wallet.rs:336]: Oh ya, that's much nicer. I had planned to add the missing function as part of my UIVK work, but I'm glad it's already been done.

**Comment by nuttycom** [comment]
> @str4d did you see @nuttycom's comment here: [#1175 (review)](https://github.com/zcash/librustzcash/pull/1175#pullrequestreview-1925276413)

This isn't something that needs to be fixed right now, because at present `zcash_client_sqlite` depends upon the `orchard` feature being enabled for `zcash_keys` (because we initially created the key types before we make things feature-flaggable, and so existing wallets will already have Orchard keys.). When in the future we do our comprehensive feature-flagging pass, then this will end up as a compilation error, and we'll correctly flag it off.

**Pull request merged**